x/slashing/client/rest/query.go

```diff
@@ -19,46 +18,98 @@ func registerQueryRoutes(cliCtx context.CLIContext, r *mux.Router, cdc *codec.Co
 		signingInfoHandlerFn(cliCtx, slashing.StoreKey, cdc),
 	).Methods("GET")
 
+	r.HandleFunc(
+		"/slashing/signing_infos",
+		signingInfoHandlerListFn(cliCtx, slashing.StoreKey, cdc),
+	).Methods("GET").Queries("page", "{page}", "limit", "{limit}")
+
 	r.HandleFunc(
 		"/slashing/parameters",
 		queryParamsHandlerFn(cdc, cliCtx),
 	).Methods("GET")
 }
 
 // http request handler to query signing info
-// nolint: unparam
 func signingInfoHandlerFn(cliCtx context.CLIContext, storeName string, cdc *codec.Codec) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		vars := mux.Vars(r)
-
 		pk, err := sdk.GetConsPubKeyBech32(vars["validatorPubKey"])
 		if err != nil {
 			rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
 			return
 		}
 
-		key := slashing.GetValidatorSigningInfoKey(sdk.ConsAddress(pk.Address()))
+		signingInfo, code, err := getSigningInfo(cliCtx, storeName, cdc, pk.Address())
 
-		res, err := cliCtx.QueryStore(key, storeName)
 		if err != nil {
-			rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
+			rest.WriteErrorResponse(w, code, err.Error())
 			return
 		}
 
-		if len(res) == 0 {
+		if code == http.StatusNoContent {
 			w.WriteHeader(http.StatusNoContent)
 			return
 		}
 
-		var signingInfo slashing.ValidatorSigningInfo
+		rest.PostProcessResponse(w, cdc, signingInfo, cliCtx.Indent)
+	}
+}
+
+// http request handler to query signing info
+func signingInfoHandlerListFn(cliCtx context.CLIContext, storeName string, cdc *codec.Codec) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var signingInfoList []slashing.ValidatorSigningInfo
+
+		_, page, limit, err := rest.ParseHTTPArgs(r)
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
+			return
+		}
+
+		// get the node
+		node, err := cliCtx.GetNode()
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
+			return
+		}
 
-		err = cdc.UnmarshalBinaryLengthPrefixed(res, &signingInfo)
+		status, err := node.Status()
```

Comment:
Why do we do an explicit status check? I don't think we do this is any other route, do we?

We should drop this if we can.

Reply:
I just needed the current height :) there is another way of getting it?

Reply:
Ohhh, sorry I missed that! Yes, there is a `GetChainHeight` in `client/rpc/block.go`. I wonder if we can leverage that?

Reply:
Much nicer, thanks!